core/types/receipt.go

```diff
@@ -117,6 +117,47 @@ func NewReceipt(root []byte, failed bool, cumulativeGasUsed uint64) *Receipt {
 	return r
 }
 
+// DeriveFields fills the receipt with their computed fields based on consensus
+// data and contextual infos like containing block and transactions.
+func (r *Receipt) DeriveFields(signer Signer, hash common.Hash, number uint64, baseFee *big.Int, blobGasPrice *big.Int, txIndex uint, gasUsed uint64, firstLogIndex uint, tx *Transaction) error {
```

Comment:
The number of parameters here is kind of scary. We need to find a better way, like passing them with a struct. Don't really have a good idea what that struct should be called.